changelog/10743.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fixed different behavior from std lib unittest of asserts with expression that contains the walrus operator in it that changes the value of a variable.
```

Comment:
I think we can update this to mean that walrus operator now is properly supported by the assertion rewriting, correct?

```suggestion
The assertion rewriting mechanism now works correctly when an assertion expression contains the walrus operator.
```

Reply:
LGTM. I pushed the changes. Thanks